### PR TITLE
Custom nnunet trainer

### PR DESCRIPTION
**Pull request by jewelltaylor**
# PR Type
[**Feature** | Fix | Documentation | Other ]

# Short Description

Small PR to add the ability to specify a custom nnunet trainer class.

### CODE REVIEW

File: fl4health/clients/nnunet_client.py [hunk 1]
```diff
@@ -141,6 +143,10 @@ def __init__(
                 provided. Defaults to None.
             reporters (Sequence[BaseReporter], optional): A sequence of FL4Health
                 reporters which the client should send data to.
+            nnunet_trainer_class (Type[nnUNetTrainer]): A nnUNetTrainer constructor.
```
Comment (emersodb) [inline fl4health/clients/nnunet_client.py:146]: Maybe add a quick note here and on the server-side that the training classes have to match each other (client and server trainers should be the same)?

File: pyproject.toml [hunk 2]
```diff
@@ -34,6 +34,7 @@ scikit-learn = "1.5.0" # Pin as it was causing issues with nnunet
 # Problematic grpcio versions cause issues, should be fixed in next flwr update
 # See https://github.com/adap/flower/pull/3853
 # https://github.com/grpc/grpc/issues/37162
+tornado = ">=6.4.2"
```
Comment (emersodb) [inline pyproject.toml:37]: Out of curiosity, why did this need upgrading?

**Review by emersodb — APPROVED** [review]
Looks good. Just two very minor comments.

File: pyproject.toml (hunk 2 above)
Comment (jewelltaylor) [reply pyproject.toml:37]: pip audit flags 6.4.1 so had to upgrade to 6.4.2 or higher

Comment (emersodb) [reply pyproject.toml:37]: Sounds good!

**Pull request merged**